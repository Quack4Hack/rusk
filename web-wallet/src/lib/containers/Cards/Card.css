.dusk-card {
  width: 100%;
}

.dusk-card__header {
  display: flex;
  gap: var(--small-gap);
  align-items: center;
  width: 100%;
}

.dusk-card__heading-container {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  width: 100%;
}

.dusk-card__heading-container--reverse {
  flex-direction: row-reverse;
  justify-content: space-between;
  gap: 0;
}

.dusk-card__header-toggle {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.dusk-card__header-controls-wrapper {
  display: flex;
  gap: var(--small-gap);
  align-items: center;
}
