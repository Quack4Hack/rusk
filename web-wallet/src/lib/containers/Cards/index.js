export { default as IconHeadingCard } from "./IconHeadingCard.svelte";
export { default as ToggleableCard } from "./ToggleableCard.svelte";
