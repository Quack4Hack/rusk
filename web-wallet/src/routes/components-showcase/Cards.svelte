<script>
  import { Card } from "$lib/dusk/components";
  import { mdiHome } from "@mdi/js";
</script>

<section>
  <Card heading="My Card" />
  <Card iconPath={mdiHome} heading="My Card" />
  <Card iconPath={mdiHome} heading="My Card">
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. In viverra
      tincidunt dolor, nec imperdiet felis tristique non. Nunc quis tempus est,
      sit amet euismod arcu. Sed malesuada scelerisque metus, ut tempor metus.
    </p>
    <p>
      Nam at nisi nibh. Nulla augue ex, imperdiet mattis commodo vel,
      ullamcorper in velit. Aenean eros dolor, condimentum sed tempor et, ornare
      in erat.
    </p>
  </Card>
</section>

<style>
  p:first-of-type {
    margin-bottom: 1em;
  }
</style>
