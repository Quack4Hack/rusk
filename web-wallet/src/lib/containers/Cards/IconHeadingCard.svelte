<script>
  import { Card, Icon } from "$lib/dusk/components";

  import "./Card.css";

  /** @type {string | undefined} */
  export let iconPath = undefined;

  /** @type {string} */
  export let heading;

  /** @type {CardGap} */
  export let gap = "default";

  /** @type {boolean} */
  export let onSurface = false;

  /** @type {boolean} */
  export let reverse = false;
</script>

<Card {...$$restProps} {gap} {onSurface}>
  <header slot="header" class="dusk-card__header">
    <div
      class="dusk-card__heading-container"
      class:dusk-card__heading-container--reverse={reverse}
    >
      {#if iconPath}
        <Icon className="dusk-card__heading-icon" path={iconPath} />
      {/if}
      <h3 class="h4">{heading}</h3>
    </div>
  </header>
  <slot />
</Card>
