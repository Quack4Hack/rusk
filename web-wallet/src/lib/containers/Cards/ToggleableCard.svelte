<script>
  import { Card, Icon, Switch } from "$lib/dusk/components";
  import { createEventDispatcher } from "svelte";

  import "./Card.css";

  /** @type {string | undefined} */
  export let iconPath = undefined;

  /** @type {string} */
  export let heading;

  /** @type {CardGap} */
  export let gap = "default";

  /** @type {boolean} */
  export let onSurface = false;

  /** @type {boolean} */
  export let isToggled = false;

  const dispatch = createEventDispatcher();

  function dispatchToggleEvent() {
    dispatch("toggle", { isToggled });
  }
</script>

<Card {...$$restProps} {gap} {onSurface} showBody={isToggled}>
  <header slot="header" class="dusk-card__header dusk-card__header-toggle">
    <h3 class="h4">{heading}</h3>
    <div class="dusk-card__header-controls-wrapper">
      <Switch
        onSurface
        bind:value={isToggled}
        on:change={dispatchToggleEvent}
      />
      {#if iconPath}
        <Icon path={iconPath} />
      {/if}
    </div>
  </header>
  {#if isToggled}
    <slot />
  {/if}
</Card>
