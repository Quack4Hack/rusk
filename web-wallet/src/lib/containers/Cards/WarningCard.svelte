<script>
  import { Card, Icon } from "$lib/dusk/components";
  import { mdiAlertOutline } from "@mdi/js";
  import "./Card.css";

  /** @type {string} */
  export let heading;

  /** @type {string | undefined} */
  export let className = undefined;

  /** @type {CardGap} */
  export let gap = "default";

  /** @type {boolean} */
  export let onSurface = false;
</script>

<Card {...$$restProps} {className} {gap} {onSurface}>
  <header slot="header" class="dusk-card__header">
    <div
      class="dusk-card__heading-container dusk-card__heading-container--warning-card"
    >
      <Icon path={mdiAlertOutline} />
      <h3 class="h4">{heading}</h3>
    </div>
  </header>
  <slot />
</Card>
