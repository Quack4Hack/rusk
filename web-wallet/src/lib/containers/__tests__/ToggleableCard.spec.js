import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { ToggleableCard } from "../Cards";

describe("IconHeadingCard", () => {
  afterEach(cleanup);

  it("renders the ToggleableCard component with a heading", () => {
    const { container } = render(ToggleableCard, {
      props: {
        heading: "My Card",
      },
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the ToggleableCard component with a heading and an icon", () => {
    const { container } = render(ToggleableCard, {
      props: {
        heading: "My Card",
        iconPath: "M3,3H21V21H3V3M5,5V19H19V5H5Z",
      },
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the ToggleableCard component with a toggle", () => {
    const { container } = render(ToggleableCard, {
      props: {
        heading: "My Card",
        iconPath: "M3,3H21V21H3V3M5,5V19H19V5H5Z",
        isToggled: true,
      },
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
