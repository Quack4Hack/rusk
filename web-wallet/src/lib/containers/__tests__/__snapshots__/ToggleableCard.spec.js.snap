// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconHeadingCard > renders the ToggleableCard component with a heading 1`] = `
<div
  class="dusk-card dusk-card--gap-default"
>
  <div
    class="dusk-card__header-container"
  >
    <header
      class="dusk-card__header"
      slot="header"
    >
      <div
        class="dusk-card__header-title"
      >
         
        <h3
          class="h4"
        >
          My Card
        </h3>
      </div>
       
      <div
        aria-checked="false"
        aria-disabled="false"
        class="dusk-switch dusk-switch--on-surface"
        role="switch"
        tabindex="0"
      />
    </header>
  </div>
   
  <div
    class="dusk-card__body-container"
  >
    
  </div>
   
</div>
`;

exports[`IconHeadingCard > renders the ToggleableCard component with a heading and an icon 1`] = `
<div
  class="dusk-card dusk-card--gap-default"
>
  <div
    class="dusk-card__header-container"
  >
    <header
      class="dusk-card__header"
      slot="header"
    >
      <div
        class="dusk-card__header-title"
      >
        <svg
          class="dusk-icon dusk-icon--size--default"
          role="graphics-symbol"
          viewBox="0 0 24 24"
        >
          <path
            d="M3,3H21V21H3V3M5,5V19H19V5H5Z"
          />
        </svg>
        
         
        <h3
          class="h4"
        >
          My Card
        </h3>
      </div>
       
      <div
        aria-checked="false"
        aria-disabled="false"
        class="dusk-switch dusk-switch--on-surface"
        role="switch"
        tabindex="0"
      />
    </header>
  </div>
   
  <div
    class="dusk-card__body-container"
  >
    
  </div>
   
</div>
`;

exports[`IconHeadingCard > renders the ToggleableCard component with a toggle 1`] = `
<div
  class="dusk-card dusk-card--gap-default"
>
  <div
    class="dusk-card__header-container"
  >
    <header
      class="dusk-card__header"
      slot="header"
    >
      <div
        class="dusk-card__header-title"
      >
        <svg
          class="dusk-icon dusk-icon--size--default"
          role="graphics-symbol"
          viewBox="0 0 24 24"
        >
          <path
            d="M3,3H21V21H3V3M5,5V19H19V5H5Z"
          />
        </svg>
        
         
        <h3
          class="h4"
        >
          My Card
        </h3>
      </div>
       
      <div
        aria-checked="true"
        aria-disabled="false"
        class="dusk-switch dusk-switch--on-surface"
        role="switch"
        tabindex="0"
      />
    </header>
  </div>
   
  <div
    class="dusk-card__body-container"
  >
    
  </div>
   
</div>
`;
